Source/OrchBase/P_builder.cpp

```diff
@@ -1146,29 +1179,65 @@ unsigned P_builder::WriteThreadVars(string& task_dir, unsigned coreNum,
   vars_cpp << "ThreadCtxt_t Thread_" << thread_num << "_Context ";              // Set the ThreadContext name
   vars_cpp << "__attribute__ ((section (\".thr" << thread_num << "_base\"))) "; // Set the target memory area
   vars_cpp << "= {";
-  vars_cpp << "&Thread_" << thread_num << "_Context,";              // VirtualAddr
   vars_cpp << "1,";                                                 // numDevTyps        
   vars_cpp << "Thread_" << thread_num << "_DeviceTypes,";           // devTyps
   vars_cpp << thread->P_devicel.size() <<  ",";                     // numDevInsts
   vars_cpp << "Thread_" << thread_num << "_Devices,";               // devInsts
   vars_cpp << ((devTyp->par->pPropsD)?"&GraphProperties,":"PNULL,");// properties
-
-  vars_cpp << "PNULL,";                                             // RTSHead              // TODO: Remove
-  vars_cpp << "PNULL,";                                             // RTSTail              // TODO: Remove
-  vars_cpp << "0,";                                                 // nextOnIdle           // TODO: Remove
-  vars_cpp << "1,";                                                 // receiveHasPriority   // TODO: Remove
-  vars_cpp << "0";                                                  // ctlEnd               // TODO: Remove
-  vars_cpp << "};\n";
   
-  /* Replacement for the above for the "new" softswitch
-  vars_cpp << ",";                              // rtsBuffSize              //TODO:
+  
+  
+  // Work out the required size for the rtsBuffSize.
+  uint32_t outputCount = 1;     // Yes, this is intentionally 1 to cope with wrapping.

```

Comment:
I feel like, between your documentation and this comment, that the `rtsBuffSize` "issue" has not been given enough attention. Could you add a longer comment here explaining our chat (last week, I think it was), where we defined this value and its bounds? :sweet_potato:

Reply:
Expanded the comment in a4b74bc